internal/component/prometheus/exporter/cloudwatch/config.go

```diff
@@ -143,6 +159,15 @@ func convertAliasesToNamespaces(a *Arguments, logger log.Logger) {
 		}
 	}
 
+	for i, job := range a.CustomNamespace {
+		if svc := yaceConf.SupportedServices.GetService(job.Namespace); svc == nil {
+			if namespace := getServiceByAlias(job.Namespace); namespace != "" {
+				level.Warn(logger).Log("msg", "service alias is deprecated, use the namespace instead", "alias", job.Namespace, "namespace", namespace)
+				a.CustomNamespace[i].Namespace = namespace
+			}
+		}
+	}
```

Comment:
The validate in the upstream repo is doing something similar except that it's returning an error if an alias is used. I guess you added this to avoid breaking changes and allow users to still run an old config?
If so, maybe you could add a comment there?

Reply:
Yep the aliases were unique to discovery jobs as discovery jobs support specific AWS services. Custom namespace jobs are meant for "everything else". Most people use them for custom metrics they send to cloudwatch but others use them to collect AWS Namespaces not supported by discovery jobs.